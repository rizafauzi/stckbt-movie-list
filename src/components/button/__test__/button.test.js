

import React from 'react'
import ReactDOM from 'react-dom'
import Button from '../button.component'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button name="Search"/> )
  expect(getByTestId('button')).toHaveTextContent('Search')
})

it('it matches snapshot', () => {
  const tree = renderer.create(<Button name="Search" />).toJSON()
  expect(tree).toMatchSnapshot()
})