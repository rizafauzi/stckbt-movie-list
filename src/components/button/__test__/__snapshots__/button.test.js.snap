// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<button
  className="sc-bdnylx UYLtV"
  data-testid="button"
>
  Search
</button>
`;
